### PR TITLE
WIP: Upgrade test

### PR DESCRIPTION
**Pull request by miketwo**
Added a test that calls `upgrade()` on all cards. Lots of failures. I could use some help fixing them all.

### CODE REVIEW

File: tests/test_cards.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+# Dynamically generate a list of all the cards.
+# Avoids interference with other tests by not using the global items.cards.
+def get_all_cards() -> list[items.Card]:
+    return [card() for card in items.Card.__subclasses__()]
+
+
+local_copy_of_cards = get_all_cards()
+
```
Comment (miketwo) [inline tests/test_cards.py:16]: You can put `@pytest.mark.only` here when running the tests locally to skip everything else

Comment (vesper-arch) [reply tests/test_cards.py:16]: Got it.

**Comment by vesper-arch** [comment]
It turns out

**Comment by vesper-arch** [comment]
I'm pretty stupid

**Comment by vesper-arch** [comment]
Untitled39_20240908112621
That's apparently not how python works

**Comment by miketwo** [comment]
I've been coding since 1987 and I still make mistakes like that. (It gets worse when you know 10+ languages - I can't keep anything straight.)  

That's why I test so much. Fixes everything.

**Comment by vesper-arch** [comment]
wonder if i can use regex to make this faster

**Comment by miketwo** [comment]
> wonder if i can use regex to make this faster

Definitely. You could do something like:
image

**Comment by vesper-arch** [comment]
I now hate vim regex

**Comment by vesper-arch** [comment]
image
HAHAHA IT WORKS YES

**Review by vesper-arch — APPROVED** [review]

**Comment by miketwo** [comment]
Huzzah! 
thumbs-up-computer-kid-gif

**Pull request merged**